client/go.mod

```diff
@@ -1,6 +1,6 @@
 module go.opentelemetry.io/collector/client
 
-go 1.23.0
```

Comment:
We should not do this, that is what @mx-psi told us :)

Reply:
Yeah if we are going to do this we should clarify our Go compatibility assurances first. This does not bring any security advantages IMHO, it just means we don't support Go versions <1.23.8

Reply:
Is it why there is no go vuln check in core? In contrib if we don't bump to 1.23.8 all those checks failed http://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/14342969178/job/40206587595

Reply:
AIUI what matters is the Go version used in CI: https://stackoverflow.com/a/73709969 so it should be resolvable by using a more recent Go version in CI

Reply:
Ack, thanks for clarifying, I reverted the go.mod files

Should I do the same in contrib? I already bumped go.mod to 1.23.8 https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39266 but it sounds like just updating CIs is enough?

Reply:
@songy23 Yeah, I think it would be better not to bump the `go` directive in contrib

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39289 for contrib